drift/core/beamtransfer.py

```diff
@@ -1191,11 +1191,13 @@ def project_vector_sky_to_telescope(self, mi, vec):
 
         vecf = np.zeros((self.nfreq, self.ntel), dtype=np.complex128)
 
-        with h5py.File(self._mfile(mi), "r") as mfile:
+        if np.any(vec != 0):
 
-            for fi in range(self.nfreq):
-                beamf = mfile["beam_m"][fi][:].reshape((self.ntel, self.nsky))
-                vecf[fi] = np.dot(beamf, vec[fi].reshape(self.nsky))
+            with h5py.File(self._mfile(mi), "r") as mfile:
+
+                for fi in range(self.nfreq):
+                    beamf = mfile["beam_m"][fi][:].reshape((self.ntel, self.nsky))
+                    vecf[fi] = np.dot(beamf, vec[fi].reshape(self.nsky))
```

Comment:
Generally it's considered a cleaner pattern to exit early with special cases, so that would be something more like simply adding:
```python
        if np.all(vec == 0):
            return vecf
```
instead of nesting all the blocks. It's a minor quibble, and much more important where you have a more extensive set of special cases/validation to do, so change it if you feel like it, but it's not needed.

Same obviously goes for the other methods.

Reply:
Legend!